CMakeLists.txt

```diff
@@ -1,189 +1,114 @@
-#or if this library should stay include only.
-
-add_library(${PROJECT_NAME} INTERFACE)
-
-if(BUILD_STATIC_LIBS)
-  add_library(${PROJECT_NAME}_static STATIC ${Vkl_All_Files})
-endif()
-
-if(BUILD_SHARED_LIBS)
-  add_library(${PROJECT_NAME}_shared SHARED ${Vkl_All_Files})
+# check if this is root
+get_directory_property(HAS_PARENT PARENT_DIRECTORY)
+
+if(NOT HAS_PARENT)
+    find_package(Git QUIET)
+    if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+    # Update submodules as needed
+        option(GIT_SUBMODULE "Check submodules during build" ON)
+        if(GIT_SUBMODULE)
+            message(STATUS "Submodule update")
+            execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
+                            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                            RESULT_VARIABLE GIT_SUBMOD_RESULT)
+            if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+                message(FATAL_ERROR "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+            endif()
+        endif()
+    endif()
 endif()
 
-# "20.1. Project Version" and "21.3. Shared Library Versioning"
-set_target_properties(${PROJECT_NAME} PROPERTIES
-  VERSION ${VulkanLaunchpad_VERSION} # VulkanLaunchpad_VERSION is automatically
-                             # set by project(VulkanLaunchpad VERSION ...)
-  SOVERSION ${VulkanLaunchpad_VERSION_MAJOR} # VulkanLaunchpad is
-                                     # automatically set by
-                                     # project(VulkanLaunchpad VERSION
-                                     # ...)
-)
-
-# Hide symbols by default as explained in chapter "21.5.1. Specifying
-# Default Visibility"
-set_target_properties(${PROJECT_NAME} PROPERTIES
-  CXX_VISIBILITY_PRESET hidden
-  VISIBILITY_INLINES_HIDDEN TRUE
-)
+set(VKL_EXTERNAL_FOLDER "Dependencies")
+set(VKL_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+set(VKL_EXTERNAL_DIR "${VKL_SOURCE_DIR}/external")
 
-# The RootNamespace of a VS Project is used to determine naming for
-# managed resource DLLs.
-set(VS_GLOBAL_ROOTNAMESPACE ${PROJECT_NAME})
+#================================#
+# Vulkan Launchpad Compiler      #
+#================================#
 
-# This sets the "keyword" attribute of the Visual Studio project.
-set_target_properties(${PROJECT_NAME} PROPERTIES
-    VS_GLOBAL_KEYWORD "Win32Proj"
-)
-
-################################################################################
-# Target name and Output directory
-################################################################################
-# ARCHIVE_OUTPUT_DIRECTORY is used for static libraries on all platforms and import libraries associated with DLLs on Windows.
-
-if(BUILD_STATIC_LIBS)
-  set_target_properties(${PROJECT_NAME}_static PROPERTIES
-    OUTPUT_NAME "${PROJECT_NAME}_static_$<CONFIG>"
-    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/${Vkl_Library_Output_Directory}"
-  )
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Does it mean that we require C++ 17? Or does it work with an older standard too?

Reply:
yes, never tried it with c++14